pkg/machine/hyperv/machine.go

```diff
@@ -769,20 +784,23 @@ func (m *HyperVMachine) startHostNetworking() (string, machine.APIForwardingStat
 
 	if logrus.IsLevelEnabled(logrus.DebugLevel) {
 		if err := logCommandToFile(c, "gvproxy.log"); err != nil {
-			return "", 0, err
+			return -1, "", 0, err
 		}
 	}
 
 	logrus.Debugf("Starting gvproxy with command: %s %v", gvproxyBinary, c.Args)
 
 	if err := c.Start(); err != nil {
-		return "", 0, fmt.Errorf("unable to execute: %s: %w", cmd.ToCmdline(), err)
+		return -1, "", 0, fmt.Errorf("unable to execute: %s: %w", cmd.ToCmdline(), err)
 	}
 
 	logrus.Debugf("Got gvproxy PID as %d", c.Process.Pid)
 
 	if len(m.MountVsocks) == 0 {
-		return forwardSock, state, nil
+		if c.Process == nil {
+			return -1, "", 0, errors.New("no valid process for gvproxy")
+		}
```

Comment:
Can this actually happen? AFAIK c.Process will be set as soon as the process is started successfully and then never can be nil. It would only be nil before c.Start() is called or if it retuned an error both of which can not happen here.

Reply:
Defensive programming - it probably never happens, but I'd prefer to avoid a possible panic.

Reply:
I am not really a fan of having impossible to hit error conditions. I know it is hard to verify that and golang certainly does not help with this but what does `no valid process for gvproxy` mean to a user? This is nothing anyone can act on and if it happens it is an internal error in our code because we use the API incorrectly.

Also this pattern is used all over the place in podman.

Reply:
Fair. I'll drop the checks.